firedrake/slate/slac/compiler.py

```diff
@@ -295,294 +548,33 @@ def compile_expression(slate_expr, tsfc_parameters=None):
-            # these expressions. However, we must create a temporary
-            # for the actee coefficient (if we haven't already).
-            actee, = exp.actee
-            if actee not in declared_temps:
-                # Declare a temporary for the coefficient
-                V = actee.function_space()
-                shape_array = [(Vi.finat_element.space_dimension(),
-                                np.prod(Vi.shape))
-                               for Vi in V.split()]
-                ctemp = ast.Symbol("auxT%d" % len(declared_temps))
-                shape = sum(n * d for (n, d) in shape_array)
-                typ = eigen_matrixbase_type(shape=(shape,))
-                aux_statements.append(ast.Decl(typ, ctemp))
-                aux_statements.append(ast.FlatBlock("%s.setZero();\n" % ctemp))
-
-                # Now we populate the temporary with the coefficient
-                # information and insert in the right place.
-                offset = 0
-                for i, shp in enumerate(shape_array):
-                    node_extent, dof_extent = shp
-                    # Now we unpack the function and insert its entries into a
-                    # 1D vector temporary
-                    isym = ast.Symbol("i1")
-                    jsym = ast.Symbol("j1")
-                    tensor_index = ast.Sum(offset,
-                                           ast.Sum(ast.Prod(dof_extent,
-                                                            isym), jsym))
-
-                    # Inner-loop running over dof_extent
-                    coeff_sym = ast.Symbol(builder.coefficient(actee)[i],
-                                           rank=(isym, jsym))
-                    coeff_temp = ast.Symbol(ctemp, rank=(tensor_index,))
-                    inner_loop = ast.For(ast.Decl("unsigned int", jsym,
-                                                  init=0),
-                                         ast.Less(jsym, dof_extent),
-                                         ast.Incr(jsym, 1),
-                                         ast.Assign(coeff_temp, coeff_sym))
-                    # Outer-loop running over node_extent
-                    loop = ast.For(ast.Decl("unsigned int", isym, init=0),
-                                   ast.Less(isym, node_extent),
-                                   ast.Incr(isym, 1),
-                                   inner_loop)
-
-                    aux_statements.append(loop)
-                    offset += node_extent * dof_extent
-
-                # Update declared temporaries with the coefficient
-                declared_temps[actee] = ctemp
-        else:
-            raise NotImplementedError(
-                "Auxiliary expr type %s not currently implemented." % type(exp)
-            )
+    # all kernel body statements must be wrapped up as a coffee.base.Block
+    assert isinstance(statements, ast.Block), (
+        "Body statements must be wrapped in an ast.Block"
+    )
```

Comment:
I still don't like all this whitespace, why not just `assert XXX, "body statements ..."`?

Reply:
Old habits... I'll go through the code and tidy it up.